File: Design-Patterns-Using-CSharp/Proxy/ProxyTest.cs
using System;

// Subject Interface
interface ISubject
{
    void Request();
}

// RealSubject Class
class RealSubject : ISubject
{
    public void Request()
    {
        Console.WriteLine("Concrete Subject Request Method");
    }
}

// Proxy Class
class Proxy : ISubject
{
    private RealSubject realSubject;

    public Proxy()
    {
        this.realSubject = new RealSubject();
    }

    public void Request()
    {
        realSubject.Request();
    }
}

// Client code
class ProxyTest
{
    static void Main(string[] args)
    {
        Proxy proxy = new Proxy();
        proxy.Request();
    }
}

/*
Concrete Subject Request Method
*/

File: Design-Patterns-Using-Java/Proxy/ProxyTest.java
interface Subject {
    void request();
}

class RealSubject implements Subject {
    @Override
    public void request() {
        System.out.println("Concrete Subject Request Method");
    }
}

class Proxy implements Subject {
    private RealSubject realSubject;

    public Proxy() {
        this.realSubject = new RealSubject();
    }

    @Override
    public void request() {
        realSubject.request();
    }
}

public class ProxyTest {
    public static void main(String[] args) {
        Proxy proxy = new Proxy();
        proxy.request();
    }
}

/* 
Concrete Subject Request Method
*/
